### ISSUE TITLE
Accessibility Testing for Veteran Facing Forms - VA Form 21-0845

### ISSUE THREAD

**Issue by jeana-adhoc** [department-of-veterans-affairs/va.gov-team#62935]
Current staging URL: https://staging.va.gov/authorization-to-disclose

### Product information

- [X] Team name, product name, and feature name have been added to the title of this issue.
- [X] Team label, product label and feature label (if applicable) have been added to this issue.

### Who completed the use of color and color contrast test?

@jeana-adhoc

### Use of color and color contrast checks

- [X] All text of 20px or smaller has a 4.5:1 contrast ratio to its background (or better)
- [X] All text of 20px or larger has a 3:1 contrast ratio to its background (or better)
- [X] Non-text elements have a 3:1  contrast ratio to their background and to neighboring elements (or better)
- [X] Color is not the only way to distinguish links from other text (eg. links are underlined)
- [X] Any charts, maps, infographics, and tables convey all information without only relying on color
- [X] Content does not refer to color, especially when providing user instructions (eg. "Click the blue button")

### How did color testing go?

No issues were uncovered

### Who completed the axe scans?

@jeana-adhoc 

### axe checks

- [X] Each page has been scanned using axe (results shared below)
- [X] axe is integrated into your end-to-end testing

### axe DevTools scan results

| Axe results  | Page |
| ------------- | ------------- |
| No issues  | [Introduction](https://staging.va.gov/supporting-forms-for-claims/substitute-claimant-form-21P-0847/introduction)  |
| No issues  | [Who is submitting this authorization](https://staging.va.gov/authorization-to-disclose/authorizer-type) |
| No issues | [Your personal information](https://staging.va.gov/authorization-to-disclose/authorizer-personal-information) |
| No issues | [Your address](https://staging.va.gov/authorization-to-disclose/authorizer-address) |
| No issues | [Your contact information](https://staging.va.gov/authorization-to-disclose/authorizer-contact-information) |
| No issues  | [Veteran personal information](https://staging.va.gov/authorization-to-disclose/veteran-personal-information)|
| No issues  | [Veteran identification information](https://staging.va.gov/authorization-to-disclose/veteran-identification-information)|
| No issues  | [Disclosure information](https://staging.va.gov/authorization-to-disclose/disclosure-information-third-party-type)|
| No issues  | [Disclosure information - person](https://staging.va.gov/authorization-to-disclose/disclosure-information-person-name)|
| No issues | [Disclosure information - organization](https://staging.va.gov/authorization-to-disclose/disclosure-information-organization-name)|
| No issues | [Disclosure information - organization representative](https://staging.va.gov/authorization-to-disclose/disclosure-information-organization-representatives)
| No issues  | [Disclosure information - address](https://staging.va.gov/authorization-to-disclose/disclosure-information-person-address)|
| No issues | [Disclosure information - organization address](https://staging.va.gov/authorization-to-disclose/disclosure-information-organization-address)
| No issues  | [Information to release](https://staging.va.gov/authorization-to-disclose/disclosure-information-scope)|
| No issues  | [Information to release - specific information](https://staging.va.gov/authorization-to-disclose/disclosure-information-limited-information)|
| No issues  | [Information to release - duration](https://staging.va.gov/authorization-to-disclose/disclosure-information-release-duration)|
| No issues | [Information to release - date](https://staging.va.gov/authorization-to-disclose/disclosure-information-release-end-date)
| No issues  | [Security question](https://staging.va.gov/authorization-to-disclose/security-information-question)|
| No issues  | [Security answer](https://staging.va.gov/authorization-to-disclose/security-information-answer)|
| No issues | [Review](https://staging.va.gov/authorization-to-disclose/review-and-submit)|
| No issues | [Confirmation](https://staging.va.gov/authorization-to-disclose/confirmation)|

### How did axe testing go?

No Axe issues were uncovered.

### Who completed the content zoom and reflow test?

@jeana-adhoc 

### Content zoom and reflow checks

- [X] All page elements are readable and usable at 200% zoom
- [X] All page elements are readable and usable at 300% zoom
- [X] All page elements are readable and usable at 400% zoom

### How did content zoom and reflow testing go?

No Issues were discovered

### Who completed the keyboard navigation test?

@jeana-adhoc 

### Keyboard navigation checks

- [X] Each link, button, form input, checkbox, radio button, select menu, and custom element can receive keyboard focus
- [X] Each link, button, form input, checkbox, radio button, select menu, and custom element responds to expected keys
- [ ] All elements under focus have a visible focus indicator
- [X] The order of [Tab] stops made sense and was appropriate for completing tasks

### How did keyboard testing go?

**Focus issue - should be fixed before launch**
**Edited to add: I think this is now fixed in the branch that's not on staging yet**
I discovered a focus issue on the [Who is submitting this authorization](https://staging.va.gov/authorization-to-disclose/authorizer-type) page. When you navigate to the first page after the introduction page, the focus is lost. With a screen reader, when you click the "Start this application without signing in" link, the browser navigates to the next page, but nothing is announced. It should announce, "Who is submitting this authorization?"  

This seems similar to when focus was lost navigating to the confirmation page. [See this ticket for more details](https://github.com/department-of-veterans-affairs/va.gov-team/issues/59616) & [Related PR](https://github.com/department-of-veterans-affairs/vets-website/pull/24425/files)

#### Dom out of order
On [Who is submitting this authorization](https://staging.va.gov/authorization-to-disclose/authorizer-type), the DOM and visual order do not match

For a screen reader user who is reading all of the content 
on the page, this will read as
"Who is submitting this authorization" as the legend, then
“Select the description that best fits you.”
Followed by “Who is submitting this authorization” (because of dom order)

https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/1a85d5c4-f9ff-4427-8a24-53db220ce5b1

DOM order should be…
1. Who is submitting…
2. Select the description that fits you.
3. I’m a Veteran
4. I’m the spouse

We should have the DOM order follow the visual flow. Also - can "Who is submitting this authorization?(*Required)" be an `<h2>` (with an h3 class on it)? This would help with wayfinding on the page.

This may be tricky because of the code involved, however, after talking with Robin, he shared that in 20-0995, he handled something similar by doing this. The most interesting thing is the `<p>` text being placed as `hint` text, and I think that would work well here.

<img width="600" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/89cf6a1e-d6b6-4ff2-97eb-e9230dfb4e5d">

#### Visual Consistency

**Making `span`s styled as `<h3>` actual `<h3>`'s **
To maintain visual and structural consistent, I think anywhere we are currently styling a legend or span to look like an `<h3>` without making it an actual `<h3>` should be rethought to see if we can indeed make it an `<h3>`.

Pages to look at:
- /authorization-to-disclose/disclosure-information-third-party-type
- /authorization-to-disclose/disclosure-information-scope
- /authorization-to-disclose/disclosure-information-limited-information
- /authorization-to-disclose/disclosure-information-release-duration
- /authorization-to-disclose/disclosure-information-release-end-date
- /authorization-to-disclose/disclosure-information-person-name
- authorization-to-disclose/disclosure-information-organization-name (Styled as h4, should be h3)
- /authorization-to-disclose/disclosure-information-organization-representatives
- /authorization-to-disclose/disclosure-information-organization-address
- /authorization-to-disclose/security-information-question

To maintain visual and structural consistency I would like to propose a change to the last form page `/authorization-to-disclose/security-information-answer`. Up until now, all of the `(* Required)` labels have been with the bitter font/h3. This last question shifts the required label down with the source sans pro font.

Currently on staging (desktop view):
<img width="539" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/c876eeeb-f7f1-4f81-97cc-a45d2e813bbc">

Proposed change (Sketch mobile view):
<img width="264" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/4baad749-77cb-43c4-b053-fd909f421d44">

And even better, would be to make the paragraph text immediately above the text input field, hint text, much like Robin's example earlier.

#### Scroll
Every page is scrolling to below the h2/progress indicator in some cases, the progress bar isn't even visible in the viewport. 

https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/233cec7b-7c42-48c1-8962-4116915d52df

**Comment by tlei123** [comment]
@jeana-adhoc  [parking here for your return from vaca...], 
I've completed the fixes in my branch for UI/UX Catchall ticket: [va-forms/21-0845/ui-ux-catchall](https://github.com/department-of-veterans-affairs/vets-website/tree/va-forms/21-0845/ui-ux-catchall)

I'm shooting for getting all that branch merged in a couple more days &mdash; I'll update here again.  Of course, in the meantime, feel free to just pull the branch and review on your local.

Thanks much for taking care of the a11y-testing!  :)

**Comment by jeana-adhoc** [comment]
Hi @tlei123 -

This is looking great. Just a couple of (hopefully) minor things.

**/authorization-to-disclose/authorizer-type**
The header, "Who is submitting this authorization? (*Required)" Is styled as an `<h3>` but it should be an `<h2>` since there is no step indicator on this page. I think you should add the `vads-u-font-size--h3` class on it, so it's sized properly.

**/authorization-to-disclose/disclosure-information-limited-information**
I think there's a stray `(* Required)` hanging out on the page. But I noticed when I typed in the text input field for "Other" the required label went away.

**/authorization-to-disclose/disclosure-information-release-end-date**
It looks like I can select a date in the past for the release end date, I think this should be a date in the future. I didn't mention it in this ticket earlier.

**/authorization-to-disclose/security-information-answer**
The `<h3>` for the security question should be inside of a `<legend>` tag, without it a screen reader only hears "Your answer" when they land on this page. It also looks like we're not using the [full web component for hint text](https://design.va.gov/storybook/?path=/docs/components-va-text-input--with-hint-text) on this question. I would remove the `hint-text` class, as it's not operating as the defined hint text.

And that's it! We're getting closer!

**Comment by tlei123** [comment]
@jeana-adhoc , final fixes done.  Please pull my branch and test locally.  Thanks much!

**Comment by tlei123** [comment]
@jeana-adhoc , pattern error-message now fixed on my local.  Changes pushed, in case you'd like to pull & validate.

![image](https://github.com/department-of-veterans-affairs/va.gov-team/assets/587583/62d955b3-ef10-448a-aa4e-b84e37508b81)

**Comment by jeana-adhoc** [comment]
@tlei123 - I've confirmed all these fixes through the browser/mac os voiceover & JAWS on windows! 

I think this can be closed now.  Thank you!!
